File: android/src/main/java/org/maplibre/reactnative/components/layers/types/raster/MLRNRasterLayer.java
package org.maplibre.reactnative.components.layers.types.raster;

import android.content.Context;

import org.maplibre.android.style.layers.RasterLayer;
import org.maplibre.reactnative.components.layers.MLRNLayer;
import org.maplibre.reactnative.components.layers.style.MLRNStyle;
import org.maplibre.reactnative.components.layers.style.MLRNStyleFactory;

public class MLRNRasterLayer extends MLRNLayer<RasterLayer> {
    public MLRNRasterLayer(Context context) {
        super(context);
    }

    @Override
    public RasterLayer makeLayer() {
        return new RasterLayer(mID, mSourceID);
    }

    @Override
    public void addStyles() {
        MLRNStyleFactory.setRasterLayerStyle(mLayer, new MLRNStyle(getContext(), mReactStyle, mMap));
    }
}

File: android/src/main/java/org/maplibre/reactnative/components/layers/types/raster/MLRNRasterLayer.kt
package org.maplibre.reactnative.components.layers.types.raster

import android.content.Context
import org.maplibre.android.style.layers.RasterLayer
import org.maplibre.reactnative.components.layers.MLRNLayer
import org.maplibre.reactnative.components.layers.style.MLRNStyle
import org.maplibre.reactnative.components.layers.style.MLRNStyleFactory

class MLRNRasterLayer(context: Context?) : MLRNLayer<RasterLayer?>(context) {
    override fun makeLayer(): RasterLayer {
        return RasterLayer(mID, mSourceID)
    }

    override fun addStyles() {
        MLRNStyleFactory.setRasterLayerStyle(mLayer, MLRNStyle(getContext(), mReactStyle, mMap))
    }
}
